### ISSUE TITLE
amp-audio and auto-advance does not work without navigational tap

### ISSUE THREAD

**Issue by Enriqe** [ampproject/amphtml#28786]
Using auto-advance + `amp-audio` does not work without a navigational tap. The progress bar will not show progress and the page won't advance after the audio is done. Clicking on the unmute button on the page containing the audio will not trigger the advancement either without a navigational tap beforehand (but the audio will play).

example:
```
      <amp-story-page id="page-0" auto-advance-after="audio0">
        <amp-story-grid-layer template="fill">
          <amp-audio autoplay id="audio0" layout="nodisplay">
            <source type="audio/mpeg" src="foo.mp3" />
          </amp-audio>
        </amp-story-grid-layer>
      </amp-story-page>

```

/cc @ampproject/wg-stories

**Comment by gmajoulet** [comment]
This one is a bit tricky as it's the default browser behavior. We might however want to fail better. One idea:
- On page navigation, if there's an audio based auto advance
- Check if autoplaying with sound is supported (create an empty audio element and try to play it, see it `play` promise resolves or rejects)
- If not supported, don't try to play the audio
- If user unmutes the story, start playing and show auto-advancement in the progress bar

**Comment by chunyahua** [comment]
So is this because the player js do not know the first page audio can be auto played?
I tried called play function (the one we already support) after the document is loaded, it does not help with auto advance. 
Also I notice that after click on the document, pages navigate to the second one and can keep autoadvancing, if then navigate back to the cover page, the navigation bar does not progress while audio is played, and the story get stuck again.

**Comment by Enriqe** [comment]
Will be fixed with #29515

**Comment by Enriqe** [comment]
Fixed by #29515
